app/models/project.rb

```diff
@@ -1,3 +1,17 @@
 class Project < ApplicationRecord
   belongs_to :challenge
+  has_many :contestant_projects
+  has_many :contestants, through: :contestant_projects
+
+  def project_challenge_theme
+    Challenge.find(self.challenge_id).theme
```

Comment:
You can make these queries simpler as well. Again, they're not wrong but they can be done more simply.
`self.challenge.theme`

This can also be called directly on the project instance in the view